### PR TITLE
Added basic animations

### LINKED ISSUE #71: Make the app interactive
The app in it's current state is a static webpage. I believe adding simple transitions and interactivity could severely overhaul the user experience.
I would start with adding [animate.css](https://daneden.github.io/animate.css/) to make transitions for elements while scrolling down.

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #71 .  
Although, what @kartik918 has suggested is to add `animate.css` and to do css-based animations which I prefer too mostly. But I came across this npm package called `react-reveal` which gives already builtin React components for basic animations. Also, we can create custom animations with it too. So, raising this PR to take opinion of all core contributors. Then we can decide whether to continue with this or to go ahead with conventional way using `animate.css`.  

Below attaching the visual of site after adding anims.
Saad Pasta Portfolio

@saadpasta and @kartik918 , please take a look at this.

### CODE REVIEW

**Comment by ashutosh1919** [comment]
@saadpasta , please help me resolve this lint error : `Found fs.link with non literal argument at index 0`

**Comment by ashutosh1919** [comment]
@saadpasta , please review.

File: .all-contributorsrc [hunk 1]
```diff
@@ -84,6 +84,15 @@
...
     },
+    {
+      "login": "ashutosh1919",
+      "name": "Ashutosh Hathidara",
+      "avatar_url": "https://avatars2.githubusercontent.com/u/20843596?s=460&u=f9348a38571fa0993d58db1c06d471b53210dc74&v=4",
+      "profile": "https://ashutosh1919.github.io",
+      "contributions": [
+        "code"
+      ]
+    },
```
Comment (kartik918) [inline .all-contributorsrc:95]: This is all-contributor bot's job so your PR shouldn't change these files as this isn't directly feature related.
Can you remove this?

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+package-lock.json
```
Comment (kartik918) [inline .gitignore:7]: Why is this added to git ignore? It's a crucial file to every project, read more [here](https://docs.npmjs.com/files/package-lock.json).

File: README.md [hunk 3]
```diff
@@ -202,6 +202,9 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
     <td align="center"><a href="https://naveen.syrusdark.website"><img src="https://avatars1.githubusercontent.com/u/49693820?v=4" width="100px;" alt=""/><br /><sub><b>Naveen M K</b></sub></a><br /><a href="https://github.com/saadpasta/developerFolio/commits?author=naveen521kk" title="Code">💻</a></td>
     <td align="center"><a href="http://msayyaf.com"><img src="https://avatars3.githubusercontent.com/u/22149734?v=4" width="100px;" alt=""/><br /><sub><b>Mohamed Sayyaf</b></sub></a><br /><a href="https://github.com/saadpasta/developerFolio/commits?author=msayyaf1" title="Documentation">📖</a></td>
   </tr>
+  <tr>
+    <td align="center"><a href="https://ashutosh1919.github.io"><img src="https://avatars2.githubusercontent.com/u/20843596?s=460&u=f9348a38571fa0993d58db1c06d471b53210dc74&v=4" width="100px;" alt=""/><br /><sub><b>Ashutosh Hathidara</b></sub></a><br /><a href="https://github.com/saadpasta/developerFolio/commits?author=ashutosh1919" title="Code">💻</a></td>
+  </tr>
```
Comment (kartik918) [inline README.md:207]: Same as above, if you find any inconsistency or error with your details, you may change it with a future PR.

File: src/components/githubRepoCard/GithubRepoCard.js [hunk 4]
```diff
@@ -8,45 +9,83 @@ export default function GithubRepoCard({ repo }) {
...
+							</span>
+						</div>
+						<div className="repo-right-stat">
+							<p>{repo.node.diskUsage} KB</p>
+						</div>
+					</div>
+				</div>
+			</div>
+		</Fade>
+	);
```
Comment (kartik918) [inline src/components/githubRepoCard/GithubRepoCard.js:90]: I see you use an auto formatter on save. To avoid unnecessary code being committed or changing an existing code style, you can turn it on per workspace instead of globally in code editor of your choice. I would suggest you to update the commits with only 3 new lines added in every file. The import statement, the opening 'react-reveal' tag and the closing tag. This makes code reviews easier and preserves existing code style.

File: src/components/header/Header.js [hunk 5]
```diff
@@ -1,44 +1,47 @@
...
+							<a href="#achievements">Achievements</a>
+						</li>
+						<li>
+							<a href="#contact">Contact Me</a>
+						</li>
+					</ul>
+				</header>
+			</div>
+		</Fade>
+	);
```
Comment (kartik918) [inline src/components/header/Header.js:45]: Same changes requested as in Github Repo Card.

File: src/containers/StartupProjects/StartupProject.js [hunk 6]
```diff
@@ -9,23 +10,32 @@ export default function StartupProject() {
...
+									</div>
+								</Fade>
+							);
+						})}
+					</div>
+					<div className="starup-project-image"></div>
+				</div>
+			</div>
+		</div>
+	);
```
Comment (kartik918) [inline src/containers/StartupProjects/StartupProject.js:40]: Same changes requested as in Github Repo Card.

File: src/containers/achievement/Achievement.js [hunk 7]
```diff
@@ -2,34 +2,43 @@ import React from "react";
...
+										footer: card.footerLink
+									}}
+								/>
+							</Fade>
+						);
+					})}
+				</div>
+			</div>
+		</div>
+	);
```
Comment (kartik918) [inline src/containers/achievement/Achievement.js:43]: Same changes requested as in Github Repo Card and for all rest of the files too.

File: src/components/footer/Footer.js [hunk 8]
```diff
@@ -1,10 +1,13 @@
...
-    <div className="footer-div">
-      <p className="footer-text">Made with ❤️ by Saad Pasta</p>
-    </div>
-  );
+		<Fade bottom duration={1000} distance="5px">
+			<div className="footer-div">
+				<p className="footer-text">Made with ❤️ by Saad Pasta</p>
+			</div>
+		</Fade>
+	);
```
Comment (kartik918) [inline src/components/footer/Footer.js:12]: Same changes requested as in Github Repo Card.

File: src/containers/skills/Skills.js [hunk 9]
```diff
@@ -2,25 +2,41 @@ import React from "react";
...
+		<div className="main" id="skills">
+			<div className="skills-main-div">
+				<Flip left duration={2000}>
+					<div className="skills-image-div">
+						<img
+							alt="Saad Working"
+							src={require("../../assests/images/developerActivity.svg")}
+						></img>
+					</div>
+				</Flip>
```
Comment (kartik918) [inline src/containers/skills/Skills.js:18]: Could you change this to Fade left? So it remains consistent with all the other transitions?

**Comment by kartik918** [comment]
@saadpasta This PR seems to have run more than half of our code through an auto formatter. 

Now generally I would be against running an existing code through an auto formatter but since this has done more than half of it already, we could update all files with formatting so in the future this doesn't happen with newer PRs.

To ensure this remains consistent for others too, I'll add configs to override user settings for Prettier (the most used code formatting extension) and run all the files through it. What do you think?

**Comment by saadpasta** [comment]
> @saadpasta This PR seems to have run more than half of our code through an auto formatter.
> 
> Now generally I would be against running an existing code through an auto formatter but since this has done more than half of it already, we could update all files with formatting so in the future this doesn't happen with newer PRs.
> 
> To ensure this remains consistent for others too, I'll add configs to override user settings for Prettier (the most used code formatting extension) and run all the files through it. What do you think?

I guess it's a good idea to add config to override the user Prettier setting but I guess running code formatting through the whole code is not a good idea. I guess we should ask @ashutosh1919  if we could remove the formatting.

**Comment by kartik918** [comment]
Yes I agree, I'll update formatting and add rules in a separate PR. For now, @ashutosh1919 could you update your PR to remove formatting and other issues addressed in the review?

**Comment by ashutosh1919** [comment]
@saadpasta and @kartik918, I agree with you and I was confused too at the time of submitting PR. But the code added by me wraps divs and other components. For example :  
```js
<div>
    <CodeComponent />
</div>
```
to
```js
<div>
    <Fade top>
    <CodeComponent />
    </Fade>
</div>
```
And I think in my opinion, it must indent in below way:   
```js
<div>
    <Fade top>
        <CodeComponent />
    </Fade>
</div>
```
So, I will change the code if that's okay for you. And +1 on adding auto formatter as @kartik918 said.

**Comment by kartik918** [comment]
Thanks for understanding, I'll be sure to add formatting rules soon to help avoid this in future contributions.

**Comment by ashutosh1919** [comment]
Closing in favor of #76.
